internal/watcher/minio.go

```diff
@@ -91,15 +99,19 @@ func (w *minioWatcher) Path() string {
 	return ""
 }
 
-func (w *minioWatcher) blpop(ctx context.Context) (*BlobEvent, error) {
-	val, err := w.client.BLPop(ctx, redisPopTimeout, w.listName).Result()
+func (w *minioWatcher) pop(ctx context.Context) (*BlobEvent, error) {
+	val, err := w.client.BLMove(ctx, w.listName, w.deadQueue, "RIGHT", "LEFT", redisPopTimeout).Result()
 	if err != nil {
 		return nil, fmt.Errorf("error retrieving from Redis list: %w", err)
 	}
 
-	event, err := w.event(val[1])
+	event, err := w.event(val)
 	if err != nil {
-		return nil, fmt.Errorf("ereror processing item received: %w", err)
+		return nil, fmt.Errorf("error processing item received: %w", err)
+	}
+
+	if _, err := w.client.LRem(ctx, w.deadQueue, 1, val).Result(); err != nil {
+		w.logger.Error(err, "Error removing message from failed list.", "list", w.deadQueue)
 	}
```

Comment:
The problem with this approach is that `w.event` is only decoding the message, it's not triggering the processing workflow. That's happening [here](https://github.com/artefactual-sdps/enduro/blob/b76841622bcb0e6b7aa9c91efa1a5ccf73210540/main.go#L272C9-L274).
This means that you're removing the message from the DLQ before you have confirmed that Temporal has started our workflow - many things could go wrong, e.g. Temporal is down, unreachable, etc... Ideally LREM is deferred for after the the workflow is started.

One possible solution would be to return a callable so the caller can call it immediately after the fact, e.g.:

```go
type ACK func() error

type Watcher interface { // ... in watcher.go
    Watch(ctx context.Context) (*BlobEvent, ACK, error)
}
```

Another alternative is what we did for STE where the watcher receives the Temporal client and knows how to start the workflow.

Reply:
@Diogenesoftoronto this is something you may want to address in a separate pull request too, after all moving away from blpop is already a step forward since that redis command is deprecated. Ensureing that the message isn't lost and is processed at least once can be done later.

Reply:
Okay. I was trying to tackle this. I will merge this if you think it is ready and open a separate pr for what I have.